cmd/mt-index-cat/out/out.go

```diff
@@ -6,14 +6,32 @@ import (
 	"strings"
 	"text/template"
 
+	"github.com/raintank/schema"
+
 	"github.com/davecgh/go-spew/spew"
 	"github.com/grafana/metrictank/idx"
 )
 
 var QueryTime int64
 
+func convertIdxMdToSchemaMd(d idx.MetricDefinition) schema.MetricDefinition {
+	md := schema.MetricDefinition{
+		Id:         d.Id,
+		OrgId:      d.OrgId,
+		Name:       d.Name.String(),
+		Interval:   d.Interval,
+		Unit:       d.Unit,
+		Mtype:      d.Mtype(),
+		Tags:       d.Tags.Strings(),
+		LastUpdate: d.LastUpdate,
+		Partition:  d.Partition,
+	}
+	md.NameWithTags() //ensure nameWithTags is set
+	return md
+}
+
```

Comment:
this should probably be a method of the `idx.MetricDefinition`

Reply:
`nameWithTags` is an unexported field in `schema.MetricDefinition` and the only way to set it is to call `NameWithTags()`.

Reply:
ok.. not sure what that has to do with my suggestion of turning this (entire function) into a method of idx.MetricDefinition ?

Reply:
Ahh it looks like you are just referring to line 29 here. Yes, that can be a method in idx.MetricDefinition.

Reply:
i am referring to the entire function declaration above my comment